File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxWater = 0;
        int left = 0;
        int right = height.size() - 1;

        while (left < right) {
            // Calculate the width and height of the container
            int width = right - left;
            int h = min(height[left], height[right]);

            // Update the maximum water if the current container can hold more
            maxWater = max(maxWater, width * h);

            // Move the pointers towards the center
            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return maxWater;
    }
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length - 1;
        int area = Integer.MIN_VALUE;
        while(left < right){
            int ht = Math.min(height[left],height[right]);
            int width = right - left;
            area = Math.max(area,ht * width);
            if(height[left] > height[right]){
                right--;
            }
            else{
                left++;
            }
        }
        return area;
    }
}